lib/ansible/modules/cloud/amazon/cloudformation_facts.py

```diff
@@ -80,6 +80,13 @@
     stack_resources: true
     stack_policy: true
 
+# Fail if the stack doesn't exist
+- name: try to get facts about a stack but fail if it doesn't exist
+  cloudformation_facts:
+    stack_name: nonexistent-stack
+    all_facts: yes
+  failed_when: cloudformation['nonexistent-stack'] is undefined
```

Comment:
I don't think this works does it? Doesn't it need  a `register` (or the `this` proposal)

Either:
```
failed_when: not this.cloudformation
```
or 
```
register: cloudformation_facts
failed_when: not cloudformation_facts.cloudformation
```

Reply:
It does work for me (I assume because cloudformation gets added to ansible_facts in this module):
```
---
- hosts: localhost
  gather_facts: no
  connection: local
  tasks:
    - name: get facts about nonexistent stack
      cloudformation_facts:
        stack_name: notthere
        all_facts: yes
        profile: shertel
        region: us-east-1
      failed_when: cloudformation['notthere'] is undefined
```
Outputs:
```
TASK [get facts about nonexistent stack] ******************************************************************************
task path: /Users/shertel/Workspace/ansible/my_playbooks/cloudformation/facts_nonex.yml:6
Using module file /Users/shertel/Workspace/ansible/lib/ansible/modules/cloud/amazon/cloudformation_facts.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: shertel
<127.0.0.1> EXEC /bin/sh -c 'echo ~ && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/shertel/.ansible/tmp/ansible-tmp-1508252420.78-238761375170325 `" && echo ansible-tmp-1508252420.78-238761375170325="` echo /Users/shertel/.ansible/tmp/ansible-tmp-1508252420.78-238761375170325 `" ) && sleep 0'
<127.0.0.1> PUT /var/folders/by/k8_fbl593dlctgqmwq5wzl2c0000gn/T/tmp32Fn1J TO /Users/shertel/.ansible/tmp/ansible-tmp-1508252420.78-238761375170325/cloudformation_facts.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /Users/shertel/.ansible/tmp/ansible-tmp-1508252420.78-238761375170325/ /Users/shertel/.ansible/tmp/ansible-tmp-1508252420.78-238761375170325/cloudformation_facts.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/Users/shertel/Workspace/ansible/venv/python2.7/bin/python /Users/shertel/.ansible/tmp/ansible-tmp-1508252420.78-238761375170325/cloudformation_facts.py; rm -rf "/Users/shertel/.ansible/tmp/ansible-tmp-1508252420.78-238761375170325/" > /dev/null 2>&1 && sleep 0'
fatal: [localhost]: FAILED! => {
    "ansible_facts": {
        "cloudformation": {}
    },
    "changed": false,
    "failed": true,
    "failed_when_result": true,
    "invocation": {
        "module_args": {
            "all_facts": true,
            "aws_access_key": null,
            "aws_secret_key": null,
            "ec2_url": null,
            "profile": "shertel",
            "region": "us-east-1",
            "security_token": null,
            "stack_events": false,
            "stack_name": "notthere",
            "stack_policy": false,
            "stack_resources": false,
            "stack_template": false,
            "validate_certs": true
        }
    }
}
	to retry, use: --limit @/Users/shertel/Workspace/ansible/my_playbooks/cloudformation/facts_nonex.retry

PLAY RECAP ************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1
```

Reply:
Oh! That's actually quite scary. Could people unexpectedly clobber their own variables through ansible_facts? 

I'm actually much more worried about ansible_facts than I was before - I really don't think it's clear that if you use a module, variables will be set whether you like it or not. I prefer explicit use of `register`.

Reply:
@willthames - you might be interested in https://github.com/ansible/ansible/pull/18445/ which is mentioned in  https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_4.rst#facts